activesupport/lib/active_support/cache.rb

```diff
@@ -307,17 +313,30 @@ def fetch(name, options = nil)
       # the cache with the given key, then that data is returned. Otherwise,
       # +nil+ is returned.
       #
+      # As with fetch, the data is only returned if it has not expired per the
+      # <tt>:expires_in<tt> option, and, if a <tt>:version</tt> parameter is passed
+      # to <tt>read</tt>, if it matches the <tt>:version</tt> it was written with.
+      #
       # Options are passed to the underlying cache implementation.
       def read(name, options = nil)
         options = merged_options(options)
-        key = normalize_key(name, options)
+        key     = normalize_key(name, options)
+
         instrument(:read, name, options) do |payload|
           entry = read_entry(key, options)
+
           if entry
             if entry.expired?
               delete_entry(key, options)
               payload[:hit] = false if payload
               nil
+            elsif entry.mismatched?(options[:version])
+              if payload
+                payload[:hit]      = false
+                payload[:mismatch] = "#{entry.version} != #{options[:version]}"
```

Comment:
Yeah, right now we're not using this information. Don't think we need it at the moment given the fact that all fragment keys are now using versions. I'll nix it.

Could you have a look at the cache hit/miss issue?